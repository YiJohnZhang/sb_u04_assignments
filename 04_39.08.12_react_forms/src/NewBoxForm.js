import React, {useState} from 'react';

function NewBoxForm({createNewBox}){

	const INITIAL_FORM_STATE = {
		width:'',
		height:'',
		backgroundColor:''
	};

	const [formState, setFormState] = useState(INITIAL_FORM_STATE);

	function handleChange(evt){

		const {name, value} = evt.target;

		setFormState((formState) => ({
			...formState,
			[name]: value
		}));


	}

	function handleSubmit(evt){

		evt.preventDefault();
		
		formState.width = Number(formState.width);
		formState.height = Number(formState.height);

		createNewBox(formState);
		setFormState(INITIAL_FORM_STATE);
		
	}

	return (

		<form>

			<label htmlFor="width">Width: </label>
			<input name="width"
				data-testid="box-newBox_input-width"
				type="number"
				value={formState.width}
				onChange={handleChange}/>

			<label htmlFor="height">Height: </label>
			<input name="height"
				data-testid="box-newBox_input-height"
				type="number"
				value={formState.height}
				onChange={handleChange}/>

			<label htmlFor="backgroundColor">Background Color: </label>
			<input name="backgroundColor"
				data-testid="box-newBox_input-bgColor"
				type="text"
				value={formState.backgroundColor}
				onChange={handleChange}/>

			<button data-testid="submitButtonName"
				onClick={handleSubmit}>Submit</button>

		</form>
	
	);

}

export default NewBoxForm;