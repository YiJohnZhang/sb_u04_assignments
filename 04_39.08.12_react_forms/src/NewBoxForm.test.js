import React from 'react';
import {render, fireEvent} from '@testing-library/react';
import NewBoxForm from './NewBoxForm';

test('smoke test', () => {

	render(<NewBoxForm />);

});

test('snapshot: empty form', () => {
	
	const {asFragment} = render(<NewBoxForm />);
	expect(asFragment()).toMatchSnapshot();

});

test('fireEvent: fields update', () => {

	// If not props, I am guess mocking or something to set the form initial state?

});

test('fireEvent: form submits', () => {

	// If not props, I am guess mocking or something to set the form initial state?

});

test('fireEvent: appends box', () => {

	// I am guess mocking or something to set the form initial state?

});