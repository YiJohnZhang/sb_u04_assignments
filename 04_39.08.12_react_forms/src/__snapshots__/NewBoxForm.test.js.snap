// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot: empty form 1`] = `
<DocumentFragment>
  <form>
    <label
      for="width"
    >
      Width: 
    </label>
    <input
      data-testid="box-newBox_input-width"
      name="width"
      type="number"
      value=""
    />
    <label
      for="height"
    >
      Height: 
    </label>
    <input
      data-testid="box-newBox_input-height"
      name="height"
      type="number"
      value=""
    />
    <label
      for="backgroundColor"
    >
      Background Color: 
    </label>
    <input
      data-testid="box-newBox_input-bgColor"
      name="backgroundColor"
      type="text"
      value=""
    />
    <button
      data-testid="submitButtonName"
    >
      Submit
    </button>
  </form>
</DocumentFragment>
`;
